status-im/status-react
# Configure Sentry for Mobile

Fixes: https://github.com/status-im/status-mobile/issues/21656

### Summary

In this PR we integrate Sentry and implement a feature flagged feature (disabled by default) to _force crash_ the app using the function `mobile/status.go#IntendedPanic()` provided by status-go.

~~PR is waiting for https://github.com/status-im/status-mobile/issues/21706 to be resolved for this one to be taken out of draft.~~

### Review notes

Documentation about Sentry in status-go: https://github.com/status-im/status-go/blob/3466ac2661bb19cd0eaa27761551de3b4d31b393/internal/sentry/README.md#L60

### Areas that may be impacted

Whenever a panic happens in status-go and usage data collection is enabled we will send error data to Sentry. In theory, nothing should change for the user, therefore nothing should be visibly different in the app's behavior.

### Steps to test

Before following the steps below, the CI must be configured with the appropriate `SENTRY_DSN_STATUS_GO` value. You can also use a free tier cloud instance to do your own testing in case you want to play around with Sentry.

1. Login with any profile.
2. Enable usage data collection (for now, Sentry and analytics are tied to the same setting in status-go).
3. Go to `Settings > Feature flags` and enable `app-monitoring > intentional-crash`
4. Go to `Settings > Advanced` > and press on `Force crash immediately`

Usually you should see the error in the Sentry dashboard in a matter of seconds.

status: ready
